src/components/Application.tsx

```diff
@@ -341,36 +362,80 @@ export class Application extends React.Component<Props, State> {
   // Internals
   //
 
-  private get detectionsForCurrentMode(): (beachfront.Job|beachfront.ProductLine)[] {
+  private updateMapMode() {
+    let mapMode: string
     switch (this.state.route.pathname) {
+      case '/create-job':
+        mapMode = this.state.bbox && this.state.searchResults ? MODE_SELECT_IMAGERY : MODE_DRAW_BBOX
+        break
       case '/create-product-line':
+        mapMode = MODE_DRAW_BBOX
+        break
       case '/product-lines':
-        return this.state.selectedFeature ? [this.state.selectedFeature as any] : this.state.productLines.records
+        mapMode = MODE_PRODUCT_LINES
+        break
       default:
-        return this.state.jobs.records.filter(j => this.state.route.jobIds.includes(j.id))
+        mapMode = MODE_NORMAL
```

Comment:
Curious as to what this was needed for? It seems nice to have it be more clear, but was there a requirements where you needed this?